### PR TITLE
OS latest version set from upgrade path

### LINKED ISSUE #3629: RFC: Major-to-major OS update only
## Context

At times OS changes requires some migration. To make migration maintainable, we typically assume upgrades from one major to the next.

In a particular case, `ova` and `generic-x86-64` updates from 6.1 and before cannot be done to 7.6 and beyond (read the details in https://github.com/home-assistant/operating-system/issues/1889).

## Decision

We should prevent upgrades which skip major versions. We do not have metadata about older release readily available, hence that does not allow us to jump to the latest version of a particular major release. However, we do know the first version of a particular major release, hence we can enforce an update to that version only, e.g.:

- `6.0` -> `7.0`
- `6.6` -> `7.0`
- `7.0` -> `8.0`

### PR DESCRIPTION
**Pull request by mdegat01**
## Proposed change

Set latest version of OS according to the update path for current major version if one is provided in the version info. Note that this PR requires https://github.com/home-assistant/version/pull/227 first.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (which adds functionality to the supervisor)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #3629
- This PR is related to issue:
- Link to documentation pull request:
- Link to cli pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast supervisor tests`)
- [x] Tests have been added to verify that the new code works.

If API endpoints of add-on configuration are added/changed:

- [ ] Documentation added/updated for [developers.home-assistant.io][docs-repository]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[docs-repository]: https://github.com/home-assistant/developers.home-assistant

### CODE REVIEW

**Review by agners — APPROVED** [review]

**Review by agners — APPROVED** [review]
The logic LGTM, unit tests seem to need addressing.

**Review by pvizeli — COMMENTED** [review]
I was not able to fix the test. Maybe it will not work? But general it's okay how we handle it

File: supervisor/updater.py [hunk 1]
```diff
@@ -243,6 +243,15 @@ async def fetch_data(self):
                 self._data[ATTR_OTA] = data["ota"]
                 if version := data["hassos"].get(self.sys_os.board):
                     events.append("os")
+                    upgrade_map = data.get("hassos-upgrade", {})
+                    if last_in_major := upgrade_map.get(str(self.sys_os.version.major)):
```
Comment (ludeeus) [inline supervisor/updater.py:247]: `.major` returns `None` for non SemVer versions (like our OS)
use `.section(0)` to get the first part of the version instead

Comment (ludeeus) [reply supervisor/updater.py:247]: Actually, that's dumb.
I'll fix that (support more variants) in the lib

Comment (ludeeus) [reply supervisor/updater.py:247]: https://github.com/home-assistant/supervisor/pull/3634

**Pull request merged**